stats/lib/platform-stats-fetchers.js

```diff
@@ -53,3 +53,27 @@ export const fetchDailyRewardTransfers = async (pgPool, filter) => {
   `, [filter.from, filter.to])
   return rows
 }
+
+/**
+ * @param {Queryable} pgPool
+ * @param {import('./typings.js').DateRangeFilter} filter
+ */
+export const fetchTopEarningParticipants = async (pgPool, filter) => {
+  const { rows } = await pgPool.query(`
+    WITH latest_scheduled_rewards AS (
+      SELECT DISTINCT ON (participant_address) participant_address, scheduled_rewards
+      FROM daily_scheduled_rewards
+      ORDER BY participant_address, day DESC
+    )
+    SELECT 
+      COALESCE(drt.to_address, lsr.participant_address) as participant_address, 
+      COALESCE(SUM(drt.amount), 0) + COALESCE(lsr.scheduled_rewards, 0) as total_rewards
```

Comment:
I think this query does not work correctly when `to` is not `now()`, because we always use the latest scheduled rewards. 

Hypothetical example:
- By yesterday, I had received 0.1 FIL via transfers, and my scheduled balance was 0.09 FIL yesterday. Total: 0.19 FIL.
- Today, I received another 0.1 FIL via transfer, and my scheduled balance was 0.02 FIL. Total: 0.22 FIL.

When I set `to=yesterday`, I believe your query will return `0.12`:
- `0.1` (transfers until yesterday)
- `0.02` (latest scheduled rewards as of today)

My conclusion: this query must always use `now()` (today) as the `to` value.

Because we depend on the `to` query param as a component of the cache key, I propose the following solution: at the top of this handler, assert that `to` is today (UTC) and return 4xx error response if it's not.

Reply:
Ah good catch, addressed in https://github.com/filecoin-station/spark-stats/pull/170/commits/633f1946b648dfc8e1480d8e426f188853f7068c

Reply:
I added a code comment to explain this constraint to future readers, see 9332282